### DISCUSSION TITLE
CS50 Check for "Go back to Bank" doesn't approve of my code, even though the pytest test_bank.py does

### DISCUSSION THREAD

**Discussion by Seine242** [community/community#64673] [Programming Help]
### Body

I wrote the code for 'Bank' in CS50 course problem set 1, and the CS50 check says that it is entirely correct, as well as when I run test_bank.py both codes seem to be right ( my output for running 'test_bank.py' is '7 PASSED').

Please consider looking at this pdf file where I explained the problem:

[Full Description Of the Issue I get.pdf](https://github.com/community/community/files/12414508/Full.Discription.Of.the.Issue.I.get.pdf)

Thank you!
-----------------------------------------------------------------------------------------------------------------------------------
Here is my code: 

when I run the CS50 check 'check50 cs50/problems/2022/python/tests/bank' it outputs 4 error messages:

( test_bank.py exist
Log
checking that test_bank.py exists...
:( correct bank.py passes all test_bank checks
Cause
expected exit code 0, not 2
Log
running pytest test_bank.py...
checking that program exited with status 0...
:| test_bank catches bank.py with incorrect values
Cause
can't check until a frown turns upside down
:| test_bank catches bank.py without case-insensitivity
Cause
can't check until a frown turns upside down
:| test_bank catches bank.py not allowing for entire phrase
Cause
can't check until a frown turns upside down):
bank.py:

def main():
greeting = str(input("Greeting: "))
print('$' + value(greeting))

def value(greeting):
hello_list = (greeting.lower()).strip().split(' ')
if 'hello' in hello_list[0]:
return "0"
elif (greeting.strip())[0].lower() == "h" and not('hello' in hello_list[0].lower()):
return "20"
else:
return "100"

if name == "main":
main()

test_bank.py:

from bank import value
import pytest

def test_int():
assert value('1') == '100'

def test_blank():
with pytest.raises(IndexError):
value('')

def test_space():
with pytest.raises(IndexError):
value(' ')

def test_0():
assert value('Hello, how"s going?') == '0'
assert value('HELLO') == '0'

def test_20():
assert value('h') == '20'
assert value('hey, good') == '20'
assert value('HOLA ') == '20'

def test_100():
assert value('wow, hello') == '100'
assert value('LalhalHa') == '100'

def test_punct():
assert value(' .,;: "') == '100'

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by queenofcorgis** [comment]
Hi @Seine242 ,

The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was closed because it contained content that violates our Community Code of Conduct, including posting the same comment in multiple places. As a result, we have hidden the duplicate comments, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).
